source/binary.cpp

```diff
@@ -850,6 +852,15 @@ void Parser::recordNumberType(size_t inst_offset,
     } else if (spv::Op::OpTypeFloat == opcode) {
       info.type = SPV_NUMBER_FLOATING;
       info.bit_width = peekAt(inst_offset + 2);
+      if (inst->num_words >= 4) {
+        const char* encoding = grammar_.lookupOperandName(
```

Comment:
This should not use string operations.

Please make a utility function to convert from  spv::FPEncoding (from spirv.hpp11) to spv_encoding_t.  It should go in source/operand.h probably.

There is an API to look up by operand type and enum value.
See  https://github.com/KhronosGroup/SPIRV-Tools/blob/7dda3c01fb4c0f9941d3cb792947d57d896ac55f/source/table2.h#L241  

(FYI: going through the grammar_ is the legacy path.)

```cpp
spvtools::OperandDesc* desc;
spv_result_t status = spvtools::LookupOperand(SPV_OPERAND_TYPE_FPENCODING, peekAt(inst_offset + 3), &desc);
if (status == SPV_SUCCESS) {
   .. the desc variable is populated.
} else {
   ... failure path. The operand is invalid with respect to what is in spirv-headers
} 
```